src/main/java/com/mcmoddev/alt/AdditionalLootTables.java

```diff
@@ -65,15 +65,21 @@ public AdditionalLootTables() {
 		public static void onCommonSetup(FMLCommonSetupEvent event) {
 			pluginLoader.loadResources();
 		}
-		
-		// TODO: fix inter-mod communication
+
+		// TODO: check and test inter-mod communication
 		@SubscribeEvent
-	    public void onIMC(InterModProcessEvent event) {
+	    public void onIMCProcess(InterModProcessEvent event) {
+			/*
+			 * Enqueue an IMC named "register" and the resource location with your MODID as domain name
+			 * net.minecraftforge.fml.InterModComms.sendTo(Your_MODID, "register", ()->new ResourceLocation(Your_MODID, "anything"));
+			 */
 	    	event.getIMCStream()
-//	    	.filter(message -> message.getMethod().equalsIgnoreCase("register"))
+	    	.filter(message -> message.getMethod().equalsIgnoreCase("register"))
 	        .forEach(message -> {
-//	            ResourceLocation value = message.getResourceLocationValue();
-//	            ALTFileUtils.copyFromResourceIfNotPresentFixed(value);
+	        	Object value = message.getMessageSupplier().get();
+	        	if (value instanceof ResourceLocation) {
+		            ALTFileUtils.copyFromResourceIfNotPresentFixed((ResourceLocation) value);

```

Comment:
hrm... not bad - I've got some ideas of how to extend this, as we could have a couple different types of message - one that registers a single file, one that registers a resource path and one that registers a "full plugin". The last two would lead to the complete removal of the plugin system - which I think is the correct direction.

Reply:
The message consist of a name and an object supplier, the name is just a string which identifies the message type, the object supplier provides the payload. The payload can be any type, there's no limitaion on this. We could have something like "register" for resourcepaths registration, "register_plugin" for ALTPlugin registration.
I would recommend that we should keep the PluginLoader, keep it there and @deprecated it. To maintain some backward compatibility, until all api users shift to the new plugin registration system.

Reply:
sounds good to me.

Reply:
Good plan